src/Microsoft.ML.Transforms/NAIndicatorTransform.cs

```diff
@@ -385,5 +475,272 @@ private void FillValues(int srcLength, ref VBuffer<bool> dst, List<int> indices,
+                    };
+            }
+
+            /// <summary>
+            /// Getter generator for vector valued inputs.
+            /// </summary>
+            private ValueGetter<VBuffer<bool>> ComposeGetterVec(IRow input, int iinfo)
+                => Utils.MarshalInvoke(ComposeGetterVec<int>, _infos[iinfo].TypeSrc.ItemType.RawType, input, iinfo);
+
+            private ValueGetter<VBuffer<bool>> ComposeGetterVec<T>(IRow input, int iinfo)
+            {
+                var getSrc = input.GetGetter<VBuffer<T>>(ColMapNewToOld[iinfo]);
+                var isNA = (RefPredicate<T>)_isNAs[iinfo];
+                var val = default(T);
+                var defaultIsNA = isNA(ref val);
+                var src = default(VBuffer<T>);
+                var indices = new List<int>();
+
+                ValueGetter<VBuffer<bool>> getter;
+
+                return getter =
+                    (ref VBuffer<bool> dst) =>
+                    {
+                        // Sense indicates if the values added to the indices list represent NAs or non-NAs.
+                        bool sense;
+                        getSrc(ref src);
+                        _parent.FindNAs(ref src, isNA, defaultIsNA, indices, out sense);
+                        _parent.FillValues(src.Length, ref dst, indices, sense);
+                    };
+            }
+        }
+    }
+
+    public sealed class NAIndicatorEstimator : IEstimator<NAIndicatorTransform>
+    {
+        private readonly IHost _host;
+        private readonly NAIndicatorTransform.ColumnInfo[] _columns;
+
+        public NAIndicatorEstimator(IHostEnvironment env, string name, string source = null)
+            : this(env, new NAIndicatorTransform.ColumnInfo(source ?? name, name))
+        {
+        }
+
+        public NAIndicatorEstimator(IHostEnvironment env, params NAIndicatorTransform.ColumnInfo[] columns)
+        {
+            Contracts.CheckValue(env, nameof(env));
+            _host = env.Register(nameof(NAIndicatorEstimator));
+            _columns = columns;
+        }
+
+        public SchemaShape GetOutputSchema(SchemaShape inputSchema)
+        {
+            _host.CheckValue(inputSchema, nameof(inputSchema));
+            var result = inputSchema.Columns.ToDictionary(x => x.Name);
+            foreach (var colInfo in _columns)
+            {
+                if (!inputSchema.TryFindColumn(colInfo.Input, out var col))
+                    throw _host.ExceptSchemaMismatch(nameof(inputSchema), "input", colInfo.Input);
+                string reason = NAIndicatorTransform.TestType(col.ItemType);
+                if (reason != null)
+                    throw _host.ExceptParam(nameof(inputSchema), reason);
+                ColumnType type = !col.ItemType.IsVector ? (ColumnType) BoolType.Instance : new VectorType(BoolType.Instance, col.ItemType.AsVector);
+                result[colInfo.Output] = new SchemaShape.Column(colInfo.Output, col.Kind, type, false, default);
```

Comment:
>default [](start = 103, length = 7)

`null` for nullables. Again, you will need `IsNormalized` here instead of `null`